Data_Package_Documentation/IGSN_Formatting/recreate_SESAR_template.R

```diff
@@ -0,0 +1,393 @@
+samples_template <- rename_column_headers(samples_template, output_sample_columns)
+
+# remove cols where all values are NA
+output_samples_template <- samples_template %>% 
+  select_if(~ !all(is.na(.))) %>% # only selects columns that have content in them
+
+  # make additional edits based on ESS-DIVE's preferences
+  # fix date: convert date time to date
+  mutate(`Collection date` = as.Date(`Collection date`)) %>% 
+  
+  # fix coordinates: round to 5 decimal places to provide accuracy within 1 meter
+  mutate(Latitude = round(Latitude, 5),
+         Longitude = round(Longitude, 5)) %>% 
+  
+  # edit Parent IGSN col by removing the full DOI link
+  mutate(`Parent IGSN` = str_remove_all(`Parent IGSN`, "https://doi.org/")) %>% 
+  
+  # select and reorder columns (reorder based on this list: https://github.com/ess-dive-community/essdive-sample-id-metadata/blob/main/guide.md)
+  select(`Sample Name`, IGSN, `Parent IGSN`,
+         Material, `Sample Type`, 
+         `Collector/Chief Scientist`, `Collection date`, `Collection method`, `Collection method description`, `Field program/cruise`,
+         Latitude, Longitude, `Primary physiographic feature`, Country,
+         `Release Date`, `Current Registrant Name`, `Original Registrant Name`, URL)
+
+
+### Check for ESS-DIVE required columns ########################################
+# checks that the required columns (according to https://ess-dive.gitbook.io/sample-id-and-metadata/guide) are present
+
+test_that("Header rows are correct", {
+  
+  # check samples
+  # first 3 cols are Object Type:, Individual Sample, User Code:
+  # 4th col is either IEWDR or IEPRS
+  expect_equal(colnames(output_sample_header)[1:3], c("Object Type:", "Individual Sample", "User Code:"))
+  expect_true(colnames(output_sample_header[4]) %in% c("IEWDR", "IEPRS")) # 'IEWDR' is for WHONDRS  'IEPRS' is not for WHONDRS
+  
+  # check sites
+  # first 3 cols are Object Type:, Site, User Code:
+  # 4th col is either IEWDR or IEPRS
+  expect_equal(colnames(output_site_header)[1:3], c("Object Type:", "Site", "User Code:"))
+  expect_true(colnames(output_site_header[4]) %in% c("IEWDR", "IEPRS")) # 'IEWDR' is for WHONDRS  'IEPRS' is not for WHONDRS
+  
+})
+
+
+test_that("Required SITES columns are present", {
+  
+  required_cols <- c("Sample Name", "Material", "Collector/Chief Scientist", 
+                     "Collection date", "Collection method description", "Field program/cruise",
+                     "Latitude", "Longitude", "Primary physiographic feature", "Release Date")
+  
+  expect_true(all(required_cols %in% colnames(output_samples_template)))
+  
+  cat("Required cols NOT in your template: ")
+  cat("\n")
+  print(setdiff(required_cols, colnames(output_samples_template)))
+  cat("\n")
+  cat("Non-required cols in your template: ")
+  cat("\n")
+  print(setdiff(colnames(output_samples_template), required_cols))
+  cat("\n")
+  cat("\n")
+
+})
+
+
+test_that("Required SAMPLES columns are present", {
```

Comment:
test failed; would be good if there were further instruction on this

Reply:
leave comment to explain context about what happens when it fails/why it fails

Reply:
commented these test lines out for now. waiting for more direction from ess-dive to determine what tests we want here